src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.MovieDao;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.Movie;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import javax.persistence.criteria.CriteriaQuery;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Enable  to save movie entity", e);
```

Comment:
```suggestion
            throw new RuntimeException("Unable  to save movie entity", e);
```
я підозрюю ти це мала на увазі🙂